### PR TITLE
DDF-2887 added catalog framework to templated docs

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

Adds to the jbake templates a section for the Catalog framework / Developing sections.

#### Who is reviewing it? 

@mcalcote @vinamartin @emmberk @Lambeaux @jrnorth @ahoffer 

#### Choose 2 committers to review/merge the PR. 

@clockard
@coyotesqrl
@lessarderic
@pklinef
@ricklarsen - Documentation
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

Template includes all relevant sections from static documentation. NOTE: "new" .adoc files are existing content with header material added. Content Review is OPTIONAL for the scope of this PR.

#### What are the relevant tickets?

[DDF-2887](https://codice.atlassian.net/browse/DDF-2887)

#### Screenshots (if appropriate)

[jdocumentation.pdf](https://github.com/codice/ddf/files/1188793/jdocumentation.pdf)

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/jdocs/content/_developingComponents/global-attribute-validators-definition-contents.adoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
+:title: Developing Global Attribute Validators
+:type: developingComponent
+:status: published
+:link: _developing_global_attribute_validators
+:summary: Creating a custom global attribute validatore.
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_developingComponents/global-attribute-validators-definition-contents.adoc]: validator

File: distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-api-contents.adoc [hunk 2]
```diff
@@ -0,0 +1,227 @@
+:title: Catalog API
+:type: catalogFrameworkIntro
+:status: published
+:summary: Introduction to Catalog API
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-api-contents.adoc]: period at the end of this summary?

File: distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-architecture-contents.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
+:title: Catalog Architecture
+:type: catalogFrameworkIntro
+:status: published
+:summary: Introduction to Catalog Architecture
```
Comment (mcalcote) [inline distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-architecture-contents.adoc]: `.`

File: distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-api-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-api-contents.adoc]: added

File: distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-architecture-contents.adoc (hunk 3 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-architecture-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_developingComponents/global-attribute-validators-definition-contents.adoc (hunk 1 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_developingComponents/global-attribute-validators-definition-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-architecture-contents.adoc (hunk 3 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-architecture-contents.adoc]: fixed.

File: distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-api-contents.adoc (hunk 2 above)
Comment (ricklarsen) [inline distribution/docs/src/main/jdocs/content/_catalog-frameworks/catalog-api-contents.adoc]: fixed.

**Comment by ricklarsen** [comment]
build now

**Pull request merged**